snapshot/storage/ql/ql.go

```diff
@@ -0,0 +1,358 @@
+package ql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"path/filepath"
+
+	"github.com/docker/containerd/snapshot"
+	"github.com/docker/containerd/snapshot/storage"
+	"github.com/pkg/errors"
+
+	_ "github.com/cznic/ql/driver"
+)
+
+type qlMetastore struct {
+	dbfile string
+}
+
+// NewMetaStore returns a snapshot MetaStore for storage of metadata related to
+// a snapshot driver backed by a ql file database. This implementation is
+// strongly consistent and does all metadata changes in a transaction to prevent
+// against process crashes causing inconsistent metadata state.
+//
+// To view mtastore with a utility use https://godoc.org/github.com/cznic/ql/ql
+func NewMetaStore(ctx context.Context, dbfile string) (storage.MetaStore, error) {
+	ms := &qlMetastore{
+		dbfile: dbfile,
+	}
+	schema := []string{
```

Comment:
Might be good to add a TODO about schema migrations in here.  We'll need to settle on that before this implementation really moves beyond a prototype.